test/assert.js

```diff
@@ -120,6 +120,10 @@ test('.not()', function (t) {
 });
 
 test('.deepEqual()', function (t) {
+	t.throws(function () {
+		assert.deepEqual({a: false}, {a: 0});
+	});
```

Comment:
A test to make sure we ignore constructor equality would be a good one. I
goofed up and never added that last time.

On Sat, Apr 23, 2016, 2:37 PM Sindre Sorhus <notifications@github.com>
wrote:

> In test/assert.js
> <https://github.com/sindresorhus/ava/pull/777#discussion_r60833744>:
>
> > @@ -120,6 +120,10 @@ test('.not()', function (t) {
> >  });
> >
> >  test('.deepEqual()', function (t) {
> > +	t.throws(function () {
> > +		assert.deepEqual({a: false}, {a: 0});
> > +	});
>
> Sure, that, and maybe some tests from
> https://github.com/sotojuan/not-so-shallow Just to make sure we never
> regress, even if we were to replace your lib with something else any time
> in the future (hopefully we won't though).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/sindresorhus/ava/pull/777/files/43e64dcb23886dc0657e4136d6be318ddcfdcdd3#r60833744>
>